src/torchmetrics/utilities/plot.py

```diff
@@ -312,8 +313,18 @@ def plot_curve(
     elif (isinstance(x, list) and isinstance(y, list)) or (
         isinstance(x, Tensor) and isinstance(y, Tensor) and x.ndim == 2 and y.ndim == 2
     ):
+        n_classes = len(x)
+        if labels is not None and len(labels) != n_classes:
+            raise ValueError(
+                "Expected number of elements in arg `labels` to match number of labels in roc curves but "
+                f"got {len(labels)} and {n_classes}"
+            )
+
         for i, (x_, y_) in enumerate(zip(x, y)):
-            label = f"{legend_name}_{i}" if legend_name is not None else str(i)
+            if labels is None:
+                label = f"{legend_name}_{i}" if legend_name is not None else str(i)
+            else:
+                label = labels[i]
```

Comment:
in case this is a list of integers we need to convert it to string first
```suggestion
                label = str(labels[i])
```